packages/block-library/src/query-pagination/index.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * Server-side rendering of the `core/query-pagination` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/query-pagination` block on the server.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the pagination for the query.
+ */
+function render_block_core_query_pagination( $attributes, $content, $block ) {
+	$page_key = 'query-' . $block->context['id'] . '-page';
+	$page     = empty( $_GET[ $page_key ] ) ? 1 : $_GET[ $page_key ];
+
+	$content = '';
+	if ( 1 !== $page ) {
+		$content .= sprintf(
+			'<div><a href="%s">Previous</a></div>',
```

Comment:
"Next" and "Previous" strings should be translatable _and_ customizable in the editor.

Reply:
I made it translatable, but customization should be done in follow-ups.